govcd/media.go

```diff
@@ -747,3 +747,90 @@ func (vdc *Vdc) QueryAllMedia(mediaName string) ([]*MediaRecord, error) {
 	util.Logger.Printf("[TRACE] Found media records by name: %#v \n", mediaResults)
 	return newMediaRecords, nil
 }
+
+// enableDownload prepares a media item for download and returns a download link
+func (media *Media) enableDownload() (string, error) {
```

Comment:
Is it a toggle switch or an idempotent function? If it's a switch, maybe it could be changed into something like `EnableDownload(enable bool)`? If not, maybe we could name it like `getDownloadLink` or something similar??

Reply:
It's idempotent. As explained in line 757, when we request a download, a task operation starts, and a file is saved somewhere. There is an ID pointing to that file added to the Media structure.

If we repeat the operation some time later, the same ID is returned.
The best name would probably be "prepareDownload". I have left as it is for two reasons:
1. it is a private function, and won't confuse users;
2. if we decide to implement downloads for vApps and vApp templates, it will be easier to find this code and use it to create something similar. All three use similar calls to prepare/enable the download

Reply:
Thanks for the explanation!